test/unit/visitor/semantic_analysis.cpp

```diff
@@ -105,3 +107,18 @@ SCENARIO("Destructor block", "[visitor][semantic_analysis]") {
         }
     }
 }
+
+SCENARIO("Ion variable in CONSTANT block", "[visitor][semantic_analysis]") {
+    GIVEN("A mod file with ion variable redeclared in a CONSTANT block") {
+        std::string nmodl_text = R"(
+            NEURON {
+                SUFFIX cdp4Nsp
+                USEION ca READ cao, cai, ica WRITE cai
+            }
+            CONSTANT { cao = 2  (mM) }
+        )";
+        THEN("Semantic analysis fails") {
+            REQUIRE(run_semantic_analysis_visitor(nmodl_text));
+        }
+    }
+}
```

Comment:
Nitpick.. New line missing at the end of the file

Reply:
note that new clang-format (`./cmake/hpc-coding-conventions/bin/format`) removed the newlines automatically.